scripts/ipfs-pi-stream/process-stream.sh

```diff
@@ -0,0 +1,112 @@
+rm -rf $currentpath
+mkdir $currentpath
+sudo mount -t tmpfs tmpfs $currentpath
+cd $currentpath
+
+what="$(date +%Y%m%d%H%M)-LIVE"
+
+# Start ffmpeg in background
+startFFmpeg &
+
+while true; do
+  nextfile=$(ls -tr ${what}*.ts 2>/dev/null | tail -n 1)
+
+  if ! [ -z "${nextfile}" ]; then
+    # Check if the next file on the list is still being written to by ffmpeg
+    if ! [ -z "$(lsof ${nextfile} | grep ffmpeg)" ]; then
+      # Wait for file to finish writing
+      # If not finished in 45 seconds something is wrong, timeout
+      inotifywait -e close_write $nextfile -t 45
+    fi
+
+    # Grab the timecode from the m3u8 file so we can add it to the log
+    timecode=`grep -B1 ${nextfile} ${what}.m3u8 | head -n1 | awk -F : '{print $2}' | tr -d ,`
+    attempts=5
+    until [[ "${timecode}" || ${attempts} -eq 0 ]]; do
+      # Wait and retry
+      sleep 0.5
+      timecode=`grep -B1 ${nextfile} ${what}.m3u8 | head -n1 | awk -F : '{print $2}' | tr -d ,`
+      attempts=$((attempts-1))
+    done
+
+    if ! [[ "${timecode}" ]]; then
+      # Set approximate timecode
+      timecode="${HLS_TIME}.000000"
+    fi
+    
+    reset_stream="$(cat ~/stream-reset)"
+    reset_stream_marker=''
+    if [[ "${reset_stream}" -eq '1' ]]; then
+      reset_stream_marker=" #EXT-X-DISCONTINUITY"
+    fi
+    
+    echo 0 > ~/stream-reset
+    # Current UTC date for the log
+    time=`date "+%F-%H-%M-%S"`
+
+    # Add ts file to IPFS
+    ret=`ipfs add ${nextfile} 2>/dev/null > ~/tmp.txt; echo $?`
+    attempts=5
+    until [[ ${ret} -eq 0 || ${attempts} -eq 0 ]]; do
+      # Wait and retry
+      sleep 1
+      ret=`ipfs add ${nextfile} 2>/dev/null > ~/tmp.txt; echo $?`
+      attempts=$((attempts-1))
+    done
+    if [[ ${ret} -eq 0 ]]; then
+      # Update the log with the future name (hash already there)
+      echo $(cat ~/tmp.txt) ${time}.ts ${timecode}${reset_stream_marker} >> ~/process-stream.log
+
+      # Remove nextfile and tmp.txt
+      rm -f ${nextfile} ~/tmp.txt
+
+      # Write the m3u8 file with the new IPFS hashes from the log
+      totalLines="$(wc -l ~/process-stream.log | awk '{print $1}')"
+      
+      sequence=0
+      if (( "${totalLines}" > ${M3U8_SIZE} )); then
+          sequence=`expr ${totalLines} - ${M3U8_SIZE}`
+      fi
+      echo "#EXTM3U" > current.m3u8
+      echo "#EXT-X-VERSION:3" >> current.m3u8
+      echo "#EXT-X-TARGETDURATION:${HLS_TIME}" >> current.m3u8
+      echo "#EXT-X-MEDIA-SEQUENCE:${sequence}" >> current.m3u8
+      tail -n ${M3U8_SIZE} ~/process-stream.log | awk '{print $6"#EXTINF:"$5",\n'${IPFS_GATEWAY}'/ipfs/"$2}' | sed 's/#EXT-X-DISCONTINUITY#/#EXT-X-DISCONTINUITY\n#/g' >> current.m3u8
+
+      # Add m3u8 file to IPFS and IPNS publish
+      m3u8hash=$(ipfs add current.m3u8 | awk '{print $2}')
+      ipfs name publish --timeout=5s $m3u8hash &
+
+      # Copy files to web server
+      cp current.m3u8 /var/www/html/live.m3u8
+      cp ~/process-stream.log /var/www/html/live.log
```

Comment:
The installation process for IPFS removes the default nginx config, and just has the IPFS reverse proxy one. This means that these files being copied over won't be served, and even if the default config wasn't removed, you'd still have the reverse proxy competing for port 80 with the default config.

IPFS should use a different port, and the default config should remain if you want the files to be able to be served.

Reply:
Issue is in a new share model of nginx in another pull request.

Reply:
The ssb-web one right? I just saw that after I published the review.

Reply:
:+1:

Reply:
Even though ssb-web solves this, this will still have to be changed to work with the new model so that these files can be distributed.